### PR TITLE
STOR-1311: Add artifacts to create bundle

### PR DESCRIPTION
**Pull request by dobsonj**
https://issues.redhat.com/browse/STOR-1311

- Update README: add OLM, remove unnecessary environment variables
- Remove config informer from NodeServiceController
- Add artifacts to create bundle
- bump openshift/library-go and openshift/api

cc @openshift/storage

### CODE REVIEW

**Comment by dobsonj** [comment]
I think I am done with planned changes for this PR, the OLM bundle generated via `create-bundle` seems to be working okay now.

File: README.md [hunk 1]
```diff
@@ -23,15 +23,22 @@ EOF
 make
 
 # Set the environment variables
-export DRIVER_IMAGE=registry.k8s.io/csi-secrets-store/driver:v1.3.3
 export OPERATOR_NAME=secrets-store-csi-driver-operator
-export PROVISIONER_IMAGE=quay.io/openshift/origin-csi-external-provisioner:latest
-export RESIZER_IMAGE=quay.io/openshift/origin-csi-external-resizer:latest
-export SNAPSHOTTER_IMAGE=quay.io/openshift/origin-csi-external-snapshotter:latest
+export DRIVER_IMAGE=registry.k8s.io/csi-secrets-store/driver:v1.3.3
```
Comment (mpatlasov) [inline README.md:27]: Would it be better to have
```
export DRIVER_IMAGE=quay.io/openshift/origin-secrets-store-csi-driver:latest
```
there?

File: config/manifests/preview/secrets-store-csi-driver-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,366 @@
...
+  - type: SingleNamespace
+    supported: false
+  - type: MultiNamespace
+    supported: false
+  - type: AllNamespaces
+    supported: true
+  install:
+    strategy: deployment
+    spec:
+      permissions:
```
Comment (mpatlasov) [inline config/manifests/preview/secrets-store-csi-driver-operator.clusterserviceversion.yaml:63]: What was the source for this lengthy list of rules below `permissions:` and `clusterPermissions:`?

File: README.md (hunk 1 above)
Comment (dobsonj) [reply README.md:27]: I tried this, but the pods failed to start with:
```
        message: Back-off pulling image "quay.io/openshift/origin-secrets-store-csi-driver:latest"  
        reason: ImagePullBackOff
```
So I think we should leave this as-is until [ART-7070](https://issues.redhat.com/browse/ART-7070) is complete and the first published image is available.

File: config/manifests/preview/secrets-store-csi-driver-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (dobsonj) [reply config/manifests/preview/secrets-store-csi-driver-operator.clusterserviceversion.yaml:63]: A few difference places:
- The namespace-specific permissions (those under `permissions`) are the same as existing operators like GCP Filestore and is limited to the namespace where the operator is installed (usually `openshift-cluster-csi-drivers`). We know for example the operator must create pods in the same namespace to start the driver.
- The cluster-wide permissions (those under `clusterPermissions`) come from:
  - common permissions required by library-go controllers that are used by most (or all) CSI operators. Some examples of this are ClusterCSIDriver, Leases, VolumeAttachments, and the operator-lock ConfigMap.
  - Roles, RoleBindings, ClusterRoles, and ClusterBindings which are required to grant permissions to the CSI driver.
  - permissions that the operator grants to the driver must first be granted to operator (the operator can not assign permissions that it does not have), so some permissions (like privileged SCC and secretproviderclasses) came from the permissions granted to the driver in [privileged_role.yaml](https://github.com/openshift/secrets-store-csi-driver-operator/blob/main/assets/rbac/privileged_role.yaml) and [secretproviderclasses_role.yaml](https://github.com/openshift/secrets-store-csi-driver-operator/blob/main/assets/rbac/secretproviderclasses_role.yaml).

**Pull request merged**